engine/src/main/scala/cromwell/engine/workflow/workflowstore/WorkflowStoreSubmitActor.scala

```diff
@@ -74,8 +95,10 @@ final case class WorkflowStoreSubmitActor(store: WorkflowStore, serviceRegistryA
   /**
     * Takes the workflow id and sends it over to the metadata service w/ default empty values for inputs/outputs
     */
-  private def registerSubmissionWithMetadataService(id: WorkflowId, originalSourceFiles: WorkflowSourceFilesCollection): Unit = {
-    processSource(_.clearEncryptedValues)(originalSourceFiles) foreach { sourceFiles =>
+  private def registerSubmissionWithMetadataService(
+      id: WorkflowId,
+      originalSourceFiles: WorkflowSourceFilesCollection): Future[Unit] = {
+    processSource(_.clearEncryptedValues)(originalSourceFiles) map { sourceFiles =>
```

Comment:
I'm not sure I see the advantage of a 
```
map {
...
()
}
```

compared to a 

```
foreach {
...
}
```

if we don't handle the failed case anyway